src/Domain/Carts/Http/Controllers/ClearUserCartController.php

```diff
@@ -3,20 +3,44 @@
 namespace Dystcz\LunarApi\Domain\Carts\Http\Controllers;
 
 use Dystcz\LunarApi\Controller;
-use Illuminate\Http\Response;
+use Dystcz\LunarApi\Domain\Carts\JsonApi\V1\CartSchema;
+use Dystcz\LunarApi\Domain\Carts\Models\Cart;
+use Illuminate\Contracts\Support\Responsable;
+use Illuminate\Support\Facades\App;
+use LaravelJsonApi\Contracts\Store\Store as StoreContract;
+use LaravelJsonApi\Core\Responses\DataResponse;
+use LaravelJsonApi\Laravel\Http\Requests\ResourceQuery;
 use Lunar\Facades\CartSession;
 
 class ClearUserCartController extends Controller
 {
     /**
      * Clear all items from user's cart.
      */
-    public function clear(): Response
-    {
-        // $this->authorize('delete', Cart::class);
+    public function clear(
+        CartSchema $schema,
+        Cart $cart,
+        string $resourceType
+    ): Responsable {
+        $this->authorize('view', $cart);
 
-        CartSession::clear();
+        $cart = CartSession::use($cart);
 
-        return response('', 204);
+        $cart->clear();
+
+        $cart->fresh();
```

Comment:
Wat gebeurt hier? Is het celaren zowel het verwijderen als het maken van een nieuwe cart?